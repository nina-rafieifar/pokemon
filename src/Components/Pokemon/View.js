import React from "react";
import { Details } from "./Details";
import { Evolutions } from "./Evolutions";
import "../Attributes/GeneralAttributes";
import "./style.scss";

export function Pokemon(prop) {
  return (
    <div id="pokemon" className="pokemon">
      <div className="pokemon__details">
        <Details pokemon={prop.pokemon} />
      </div>
      <div className="pokemon__general">
        <pokemon-general-attr
          name={prop.pokemon.name}
          number={prop.pokemon.number}
          image={prop.pokemon.image}
        ></pokemon-general-attr>
      </div>
      <div className="pokemon__evolutions">
        <Evolutions pokemon={prop.pokemon} />
      </div>
    </div>
  );
}
