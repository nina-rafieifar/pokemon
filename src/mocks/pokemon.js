export const pokemon = {
  attacks: { fast: [{ damage: 12, name: "Tackle" }], special: [{ damage: 70, name: "Power Whip" }] },
  evolutionRequirements: { amount: 25, name: "Bulbasaur candies" },
  evolutions: [
    {
      id: "UG9rZW1vbjowMDI=",
      number: "002",
      name: "Ivysaur",
      image: "https://img.pokemondb.net/artwork/ivysaur.jpg"
    },
    {
      id: "UG9rZW1vbjowMDM=",
      number: "003",
      name: "Venusaur",
      image: "https://img.pokemondb.net/artwork/venusaur.jpg"
    }
  ],
  height: { __typename: "PokemonDimension", minimum: "0.61m", maximum: "0.79m" },
  id: "UG9rZW1vbjowMDE=",
  image: "https://img.pokemondb.net/artwork/bulbasaur.jpg",
  name: "Bulbasaur",
  number: "001",
  resistant: ["Water", "Electric", "Grass", "Fighting", "Fairy"],
  weaknesses: ["Fire", "Ice", "Flying", "Psychic"],
  weight: { minimum: "6.04kg", maximum: "7.76kg" }
};
