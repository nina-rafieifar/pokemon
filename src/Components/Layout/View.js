import React from "react";
import Carousel from "react-material-ui-carousel";
import "./style.scss";
import { useQuery } from "@apollo/client";
import { POKEMONS } from "../Queries/Pokemons";
import { Pokemon } from "../Pokemon/View";
import Spinner from "react-spinner-material";
export function Layout() {
  const { loading, error, data } = useQuery(POKEMONS);

  return loading ? (
    <div className="layout__spinner">
      <Spinner radius={80} color={"red"} stroke={8} visible={true} />
    </div>
  ) : error ? (
    <div className="layout__error">There has been an error. Please try again later.</div>
  ) : (
    <Carousel
      className="layout layout__carousel"
      indicators={false}
      autoPlay={false}
      animation="slide"
      timeout={500}
      swipe={true}
      cycleNavigation={false}
    ></Carousel>
  );
}
