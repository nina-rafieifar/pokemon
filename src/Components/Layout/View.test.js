import React from "react";
import { render } from "@testing-library/react";
import { Layout } from "./View";
import { useQuery } from "@apollo/client";
import { pokemon } from "../../mocks/pokemon";

jest.mock("@apollo/client", () => ({
  useQuery: jest.fn(),
  gql: jest.fn()
}));

describe("hierarchy component", () => {
  const mockUseQuery = useQuery;
  let data = null;
  let error = null;
  let loading = true;

  beforeEach(() => {
    mockUseQuery.mockImplementation(() => {
      return {
        data: data,
        error: error,
        loading: loading
      };
    });
  });
  test("renders Layout", () => {
    const element = render(<Layout />);
    expect(element).toMatchSnapshot();
  });

  test("renders Layout", () => {
    data = null;
    error = "Has a weird error";
    loading = false;
    const element = render(<Layout />);
    expect(element).toMatchSnapshot();
  });

  test("renders Layout", () => {
    data = { pokemons: [pokemon] };
    error = null;
    loading = false;
    const element = render(<Layout />);
    expect(element).toMatchSnapshot();
  });
});
