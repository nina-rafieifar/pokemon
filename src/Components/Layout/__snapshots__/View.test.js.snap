// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hierarchy component renders Layout 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="layout__spinner"
      >
        <div
          class="react-spinner-material"
          style="width: 80px; height: 80px; border-color: red; border-width: 8px;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="layout__spinner"
    >
      <div
        class="react-spinner-material"
        style="width: 80px; height: 80px; border-color: red; border-width: 8px;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`hierarchy component renders Layout 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="layout__error"
      >
        There has been an error. Please try again later.
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="layout__error"
    >
      There has been an error. Please try again later.
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`hierarchy component renders Layout 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Carousel-root-1 layout layout__carousel"
      >
        <div
          class="CarouselItem"
        >
          <div
            style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 500ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 500ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
          >
            <div
              class="pokemon"
              id="pokemon"
            >
              <div
                class="pokemon__details"
              >
                <div
                  class="pokemon__details_table"
                >
                  <pokemon-abilities
                    resistant="Water,Electric,Grass,Fighting,Fairy"
                    weaknesses="Fire,Ice,Flying,Psychic"
                  />
                  <div
                    class="pokemon__details_attacks"
                  >
                    <div
                      class="chips"
                    >
                      <div
                        class="chip__text"
                      >
                        Fast
                      </div>
                      <pokemon-attacks
                        damage="12"
                        name="Tackle"
                      />
                    </div>
                    <div
                      class="chips"
                    >
                      <div
                        class="chip__text"
                      >
                        Special
                      </div>
                      <div>
                        <pokemon-attacks
                          damage="70"
                          name="Power Whip"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="pokemon__general"
              >
                <pokemon-general-attr
                  image="https://img.pokemondb.net/artwork/bulbasaur.jpg"
                  name="Bulbasaur"
                  number="001"
                />
              </div>
              <div
                class="pokemon__evolutions"
              >
                <table
                  class="pokemon__evolutions_table"
                >
                  <tbody>
                    <tr>
                      <td>
                        <div
                          class="pokemon__evolutions_container"
                        >
                          <p>
                            Evolution
                          </p>
                          <div
                            class="pokemon__evolutions_items"
                          >
                            <pokemon-evolution
                              image="https://img.pokemondb.net/artwork/bulbasaur.jpg"
                              name="Bulbasaur"
                              number="001"
                            />
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                              />
                            </svg>
                            <pokemon-evolution
                              image="https://img.pokemondb.net/artwork/ivysaur.jpg"
                              name="Ivysaur"
                              number="002"
                            />
                          </div>
                        </div>
                      </td>
                    </tr>
                    <tr>
                      <td
                        class="pokemon__details_items"
                      >
                        <pokemon-sizes
                          heightmax="0.79m"
                          heightmin="0.61m"
                          weightmax="7.76kg"
                          weightmin="6.04kg"
                        />
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Carousel-root-1 layout layout__carousel"
    >
      <div
        class="CarouselItem"
      >
        <div
          style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 500ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 500ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
        >
          <div
            class="pokemon"
            id="pokemon"
          >
            <div
              class="pokemon__details"
            >
              <div
                class="pokemon__details_table"
              >
                <pokemon-abilities
                  resistant="Water,Electric,Grass,Fighting,Fairy"
                  weaknesses="Fire,Ice,Flying,Psychic"
                />
                <div
                  class="pokemon__details_attacks"
                >
                  <div
                    class="chips"
                  >
                    <div
                      class="chip__text"
                    >
                      Fast
                    </div>
                    <pokemon-attacks
                      damage="12"
                      name="Tackle"
                    />
                  </div>
                  <div
                    class="chips"
                  >
                    <div
                      class="chip__text"
                    >
                      Special
                    </div>
                    <div>
                      <pokemon-attacks
                        damage="70"
                        name="Power Whip"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="pokemon__general"
            >
              <pokemon-general-attr
                image="https://img.pokemondb.net/artwork/bulbasaur.jpg"
                name="Bulbasaur"
                number="001"
              />
            </div>
            <div
              class="pokemon__evolutions"
            >
              <table
                class="pokemon__evolutions_table"
              >
                <tbody>
                  <tr>
                    <td>
                      <div
                        class="pokemon__evolutions_container"
                      >
                        <p>
                          Evolution
                        </p>
                        <div
                          class="pokemon__evolutions_items"
                        >
                          <pokemon-evolution
                            image="https://img.pokemondb.net/artwork/bulbasaur.jpg"
                            name="Bulbasaur"
                            number="001"
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                            />
                          </svg>
                          <pokemon-evolution
                            image="https://img.pokemondb.net/artwork/ivysaur.jpg"
                            name="Ivysaur"
                            number="002"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                  <tr>
                    <td
                      class="pokemon__details_items"
                    >
                      <pokemon-sizes
                        heightmax="0.79m"
                        heightmin="0.61m"
                        weightmax="7.76kg"
                        weightmin="6.04kg"
                      />
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
